internal/nodes/avi_model_l7_translator.go

```diff
@@ -318,6 +318,35 @@ func (o *AviObjectGraph) ConstructAviL7VsNode(vsName string, key string) *AviVsN
 	o.ConstructShardVsPGNode(vsName, key, avi_vs_meta)
 	o.ConstructHTTPDataScript(vsName, key, avi_vs_meta)
 	var fqdns []string
+	subDomains := GetDefaultSubDomain()
+	autoFQDN := true
+	if lib.GetL4FqdnFormat() == lib.AutoFQDNDisabled {
+		autoFQDN = false
+	}
+	if subDomains != nil && autoFQDN {
+		var fqdn string
+		// honour defaultSubDomain from values.yaml if specified
+		defaultSubDomain := lib.GetDomain()
+		if defaultSubDomain != "" && utils.HasElem(subDomains, defaultSubDomain) {
+			subDomains = []string{defaultSubDomain}
+		}
+
+		// subDomains[0] would either have the defaultSubDomain value
+		// or would default to the first dns subdomain it gets from the dns profile
+		subdomain := subDomains[0]
+		if strings.HasPrefix(subDomains[0], ".") {
+			subdomain = strings.Replace(subDomains[0], ".", "", -1)
+		}
+		if lib.GetL4FqdnFormat() == lib.AutoFQDNDefault {
+			// Generate the FQDN based on the logic: <svc_name>.<namespace>.<sub-domain>
+			fqdn = vsName + "." + lib.GetTenant() + "." + subdomain
+		} else if lib.GetL4FqdnFormat() == lib.AutoFQDNFlat {
+			// Generate the FQDN based on the logic: <svc_name>-<namespace>.<sub-domain>
+			fqdn = vsName + "-" + lib.GetTenant() + "." + subdomain
+		}
+		utils.AviLog.Infof("key: %s, msg: Configured the shared VS with default fqdn as: %s", fqdn)
+		fqdns = append(fqdns, fqdn)
+	}
 
 	subDomains := GetDefaultSubDomain()
 	if subDomains != nil {
```

Comment:
It seems we have overlapping code here(line number 352-359). so fqdns might be duplicate. Can you please check this?

Reply:
It appears that the branch was old and the snapshot you are pointing to you is from the older branch. This code does not exist in the current `master` branch.

Either way, I will update the branch and repost.

Reply:
I just rebased the branch, you won't see the code now :)